### PR TITLE
move the stih410 console to the generic framework

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

File: core/drivers/stih_asc.c [hunk 1]
```diff
@@ -35,14 +36,20 @@
...
+
 	while (!(read32(base + ASC_STATUS) & ASC_STATUS_TX_EMPTY))
 		;
 }
 
-void stih_asc_putc(int ch, vaddr_t base)
+static void stih_asc_putc(struct serial_chip *chip, int ch)
 {
+	struct stih_asc_pd *pd = container_of(chip, struct stih_asc_pd, chip);
+	vaddr_t base = io_pa_or_va(&pd->base);
```
Comment (jforissier) [inline core/drivers/stih_asc.c]: Minor: helper function maybe? (https://github.com/OP-TEE/optee_os/blob/master/core/drivers/pl011.c#L95-L101).

**Comment by etienne-lms** [comment]
thanks for the comment.
tag applied.

**Comment by jforissier** [comment]
Can you rebase please?

**Comment by etienne-lms** [comment]
done, thanks.

**Pull request merged**